Tickr/TickrTests/TickrTests.swift

```diff
@@ -0,0 +1,288 @@
+//  TickrTests
+//
+//  Created by Stephen Payne on 4/8/16.
+//  Copyright © 2016 Stephen Payne. All rights reserved.
+//
+
+import XCTest
+
+@testable import Tickr
+
+class TickrTests: XCTestCase {
+    let testData = TickrTestData()
+    
+    override func setUp() {
+        super.setUp()
+        //create test data for all tests to use
+        
+
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+    
+    
+    
+}
+
+class StockManagerTests: TickrTests {
+    var stocks = [Stock]()
+    var manager = StockManager.sharedInstance
+    
+    
+    //MARK: - Parsing Tests
+    func testParseJSON() {
+        do {
+            let json = try NSJSONSerialization.JSONObjectWithData(
+                TickrTestData.singleResultData!,
+                options:NSJSONReadingOptions.MutableContainers
+                ) as! NSDictionary
+            
+            let stockArrayWithData = manager.parseJSON(json)
+            XCTAssertNotNil(stockArrayWithData, "Data couldn't be parsed. Check the data file to make sure the JSON structure hasn't changed")
+            
+        } catch {
+            //the test json files have been moved or deleted
+            XCTFail("The expected data files have been moved or deleted")
+        }
+    }
+    
+    func testParseJSONWithDifferentKeys() {
+        do {
+            let json = try NSJSONSerialization.JSONObjectWithData(
+                TickrTestData.badResultData!,
+                options:NSJSONReadingOptions.MutableContainers
+                ) as! NSDictionary
+            
+            let stockArrayWithData = manager.parseJSON(json)
+            XCTAssertNil(stockArrayWithData, "The parsing should've failed" )
+        } catch {
+            //the test json files have been moved or deleted
+            XCTFail("The expected data files have been moved or deleted")
+        }
+    }
+    
+    func testParseStockDataWithSingleResult() {
+
+        do {
+            let json = try NSJSONSerialization.JSONObjectWithData(
+                TickrTestData.singleResultData!,
+                options:NSJSONReadingOptions.MutableContainers
+            ) as! NSDictionary
+            
+            if let stockArrayWithData = manager.parseJSON(json) {
+                manager.parseStockData(stockArrayWithData)
+                XCTAssertEqual(manager.stockArray.count, 1, "There should be 1 stock in the array ")
+                
+            } else {
+                XCTFail("The data couldn't be parsed. Check the data file to make sure the JSON structure hasn't been changed")
+            }
+            
+        } catch {
+            //the test json files have been moved or deleted
+            XCTFail("The expected data files have been moved or deleted")
+        }
+    }
+    
+    func testParseStockDataWithMultipleResults() {
+        
+        do {
```

Comment:
Don't do this `do/catch` logic you're duplicating work for yourself, if your assertion fails, it will throw an error, thats why you don't need to do it yourself.

Tip: Read how other good coders write unit tests in swift. See here: https://github.com/Alamofire/Alamofire/blob/master/Tests/CacheTests.swift